src/composition/runner.rs

```diff
@@ -168,6 +168,15 @@ impl SubgraphWatchers {
     }
 }
 
+/// An event denoting that the subgraph has changed, emitting its name and the SDL reflecting that
+/// change
+pub struct SubgraphChanged {
```

Comment:
A bit of a nit, but for consistency with the rest of this work: 
can this expose `name` and `sdl` as getters and expose a constructor to create this?
This will echo the notion that an event is immutable after creation. Not that we'd mutate it for any reason, but it just firms up the contract.

Reply:
yep, done

Reply:
hmm, not sure why it's not marking this as outdated, but `#[derive(Getters)]` is now there